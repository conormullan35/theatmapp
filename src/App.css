.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}
.ovrDrft{
  color:red;
}

#navBar a:visited {
  color:white;
}
#navBar a{
  color:white;
}
#navBar {
  color:white;
  text-align: center;
  position: relative;
  z-index:999;
}
.App-header {
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

body{
  padding: 0px;
  margin: 5vh 0;
  background: linear-gradient(to right, #5433FF, #20BDFF, #A5FECB);
  font-family:'BebasNeue-Regular';
}

#gradient
{
 width: 100%;
 height: 800px;
 padding: 0px;
 margin: 0px;
}

input {
  width: 100%;
  height: 100px;
  padding: 20px;
  font-size: 20px;
  border: none;
  box-sizing: border-box;
}

.numeric-theme {
  max-width: 320px;
  margin: 0 auto;
}
.gridmain{
  justify-content: center;
}

.cardBtn{
  color:white;
  -webkit-transform: rotate(90deg);
  -moz-transform: rotate(90deg);
  -o-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
  margin:auto;
  display:block;
  font-size:12em;
  display: flex;
  border-color: transparent;
  cursor:pointer;

}

@keyframes slideUp {
	0%{
		-webkit-transform: translateY(0);
		-ms-transform:     translateY(0);
		transform:         translateY(0)
	}
	100% {
		-webkit-transform: translateY(-20px);
		-ms-transform:     translateY(-20px);
		transform:         translateY(-20px)
	}

}

.cardDiv{
  margin:auto;
  z-index:-1;
  width:50%!important;
  display: flex;
  align-items: center;
  justify-content: center;
  -webkit-animation-duration: 0.3s;
  animation-duration: 0.3s;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode:forwards;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  animation-iteration-count: 1;
  -webkit-animation-iteration-count: 1;
}

.cardDiv:hover{
  animation-fill-mode: forwards;  
  animation-name: slideUp;
  -moz-animation-name: slideUp;
  border: none;
  -webkit-box-shadow: none;
  -moz-box-shadow: none;
  box-shadow: none;
}
.cardDiv:focus, button.cardBtn:focus{
    outline: none;
}
.cardDiv button.cardBtn, button.cardBtn:active{
  background-color: transparent;
  transition: color 0s;
}

.wraptb{
  margin:auto;
  display: block;
  align-items: center;
  justify-content: center;
  text-align: center;
  color:white;
  width:50%;
  height:200px;
}
.behind{
  margin:auto;
  font-size: 5vh;
  display: flex;
  align-items: center;
  justify-content: center;
  text-align: center;
  color:white;
  height:200px;
  background:inherit;
  position: relative;
  z-index:999;
  
}
.cardSlot{
  margin:auto;
  display: flex;
  top: 3%;
  position: relative;
  align-items: center;
  text-align: center;
  width:200px;
  height:20px;
  background:inherit;
  border:5px solid white;
  border-bottom: none;
  border-radius:10px;
}

.invalidinput{
  /* Start the shake animation and make the animation last for 0.5 seconds */
  animation: shake 0.5s;

  /* When the animation is finished, start again */
  animation-iteration-count: 1;
}

@keyframes shake {
  0% { transform: translate(1px, 1px) rotate(0deg); }
  10% { transform: translate(-1px, -2px) rotate(-1deg); }
  20% { transform: translate(-3px, 0px) rotate(1deg); }
  30% { transform: translate(3px, 2px) rotate(0deg); }
  40% { transform: translate(1px, -1px) rotate(1deg); }
  50% { transform: translate(-1px, 2px) rotate(-1deg); }
  60% { transform: translate(-3px, 1px) rotate(0deg); }
  70% { transform: translate(3px, 1px) rotate(-1deg); }
  80% { transform: translate(-1px, -1px) rotate(1deg); }
  90% { transform: translate(1px, 2px) rotate(0deg); }
  100% { transform: translate(1px, -2px) rotate(-1deg); }
}


.numeric-theme + .simple-keyboard-preview {
  max-width: 320px;
}

.simple-keyboard.hg-theme-default.numeric-theme
  .hg-button.hg-standardBtn.hg-button-at {
  max-width: none;
}
.app {
  display: flex;
  justify-content: center;
  align-items: center;
}

.row {
  display: flex;
  width: 100%;
}

.row .clearBtn{
  background-color: #c7b909;
}
.row .clearBtn:hover{
  color: #c7b909;
}
.row .confirmBtn{
  background-color: forestgreen;
}
.row .confirmBtn:hover{
  color: forestgreen;
}

@font-face {
  font-family: 'BebasNeue-Regular';
  src: url('./css/fonts/BebasNeue-Regular.eot'); /* IE9 Compat Modes */
  src: url('./css/fonts/BebasNeue-Regular.eot?#iefix') format('embedded-opentype'), /* IE6-IE8 */
       url('./css/fonts/BebasNeue-Regular.woff2') format('woff2'), /* Super Modern Browsers */
       url('./css/fonts/BebasNeue-Regular.woff') format('woff'), /* Pretty Modern Browsers */
       url('./css/fonts/BebasNeue-Regular.ttf')  format('truetype'), /* Safari, Android, iOS */
}

@font-face {
  font-family: 'Roboto-Light-webfont';
  src: url('./css/fonts/Roboto-Light-webfont.eot');
  src: url('./css/fonts/Roboto-Light-webfont.eot?#iefix') format('embedded-opentype')
       url('./css/fonts/Roboto-Light-webfont.woff2') format('woff2'), /* Super Modern Browsers */
       url('./css/fonts/Roboto-Light-webfont.woff') format('woff'),
       url('./css/fonts/Roboto-Light-webfont.ttf') format('truetype'),
       url('./css/fonts/Roboto-Light-webfont.svg#robotolight') format('svg');
  font-weight: normal;
  font-style: normal;

}
