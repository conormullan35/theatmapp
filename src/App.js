import React from "react"
import SplashScreen from "./components/SplashScreen"
import Dashboard from "./components/Dashboard"
import PinCode from "./components/PinCode"
import { Route, Switch, Redirect } from "react-router-dom";
import Container from '@material-ui/core/Container';


const App = (props) => {
    
    return (
        <main>
            
            <Container maxWidth="sm">

                <Switch>
                    {/* <Route path="/" component={Home} exact /> */}
                    <Route exact path="/">
                        <Redirect to="/splashscreen" component={SplashScreen} />
                    </Route>
                    <Route path="/splashscreen" component={SplashScreen} />
                    <Route path ="/dashboard" component={Dashboard} />
                    <Route path ="/pincode" component={PinCode} />
                </Switch>
            </Container>
        </main>
    )
}

// import React, { Component } from 'react';
// import Keyboard from 'react-simple-keyboard';
// import 'react-simple-keyboard/build/css/index.css';
// import { BrowserRouter, Route, Switch } from 'react-router-dom';

// class App extends Component {

//   state = {
//     layoutName: "default",
//     input: ""
//   };

//   onChange = input => {
//     this.setState({
//       input: input
//     });
//     console.log("Input changed", input);
//   };

//   onChangeInput = event => {
//     let input = event.target.value;
//     this.setState(
//       {
//         input: input
//       },
//       () => {
//         this.keyboard.setInput(input);
//       }
//     );
//   };
  
//   // onChange = (input) => {
//   //   console.log("Input changed", input);
//   // }

//   onKeyPress = (button) => {
//     console.log("Button pressed", button);
//   }

//   render(){
//     return (
//       <main>
//       <Switch>
//             </Switch>
//       </main>
//       <div>
//       <input 
//         value={this.state.input} 
//         className="input" 
//         placeholder="Tap on the virtual keyboard to start" 
//         onChange={e => this.onChangeInput(e)}/>
	
//       <Keyboard
//         onChange={this.onChange}
//         onKeyPress={this.onKeyPress}
//         theme={"hg-theme-default hg-layout-default myTheme"}
//           layoutName={this.state.layoutName}
//           layout={{
//             default: ["1 2 3", "4 5 6", "7 8 9", " 0 ", "{bksp}"],
//             shift: ["! / #", "$ % ^", "& * (", "{shift} ) +", "{bksp}"]
//           }}
//       />
//       </div>
//     );
//   }
// }
// import logo from './logo.svg';
// import './App.css';
// import React,{ useState } from 'react';
// import Person from './Person/Person'
// import Button from '@material-ui/core/Button';
// import ButtonGroup from '@material-ui/core/ButtonGroup';
// import Keyboard from "react-simple-keyboard";
// import "react-simple-keyboard/build/css/index.css";
// import "./index.css";

// const App = props =>{
//   const state = {
//     layoutName: "default",
//     input: ""
//   };

//   const onChange = input => {
//     this.setState({ input });
//     console.log("Input changed", input);
//   };

//   const onKeyPress = button => {
//     console.log("Button pressed", button);

//     /**
//      * If you want to handle the shift and caps lock buttons
//      */
//     if (button === "{shift}" || button === "{lock}") this.handleShift();
//   };

//   const handleShift = () => {
//     const layoutName = this.state.layoutName;

//     this.setState({
//       layoutName: layoutName === "default" ? "shift" : "default"
//     });
//   };

//   const onChangeInput = event => {
//     const input = event.target.value;
//     this.setState({ input });
//     this.keyboard.setInput(input);
//   };

//   const[numbval, setNumber] = useState('');
  


//   const [personsState, setPersonsState] = useState({
//       numberpad: [
//         { num: '1'},
//         { num: '2'},
//         { num: '3'},
//         { num: '4'},
//         { num: '5'},
//         { num: '6'},
//         { num: '7'},
//         { num: '8'},
//         { num: '9'},
//         { num: '0'}

//       ]});
    
//       const switchNameHanlder = () => {
//         setPersonsState({
//           persons: [
//             { name: 'Maximilian', age: 28},
//             { name: 'Manu', age: 29},
//             { name: 'Max', age: 26},
//           ]
//         })
//       }

//   // state ={
//   //   persons: [
//   //     { name: 'Max', age: 28},
//   //     { name: 'Manu', age: 29},
//   //     { name: 'Max', age: 26},
      
//   //   ],
//   //   otherState: 'some other value'
//   // }

//   // switchNameHanlder = () => {
//   //   this.setState({
//   //     persons: [
//   //       { name: 'Maximilian', age: 28},
//   //       { name: 'Manu', age: 29},
//   //       { name: 'Max', age: 26},
//   //     ]
//   //    })
//   // }
    
//     return(
//       <div className="App">
//       <header className="App-header">
//         <img src={logo} className="App-logo" alt="logo" />

//         <div>
//           <input
//             value={this.state.input}
//             placeholder={"Tap on the virtual keyboard to start"}
//             onChange={this.onChangeInput}
//           />
//           <Keyboard
//             keyboardRef={r => (this.keyboard = r)}
//             layoutName={this.state.layoutName}
//             onChange={this.onChange}
//             onKeyPress={this.onKeyPress}
//           />
//         </div>


//         <div class="simple-keyboard"></div>
//         <ButtonGroup color="primary" aria-label="outlined primary button group">
//           <Button onClick={() => { alert('One') }}>One</Button>
//           <Button onClick={() => { alert('Two') }}>Two</Button>
//           <Button onClick={() => { alert('Three') }}>Three</Button>
//         </ButtonGroup>
//         <ButtonGroup color="primary" aria-label="outlined primary button group">
//           <Button>Four</Button>
//           <Button>Five</Button>
//           <Button>Six</Button>
//         </ButtonGroup>
//         <ButtonGroup color="primary" aria-label="outlined primary button group">
//           <Button>Seven</Button>
//           <Button>Eight</Button>
//           <Button>Nine</Button>
//         </ButtonGroup>
//         <ButtonGroup color="primary" aria-label="outlined primary button group">
//           <Button>Zero</Button>
//         </ButtonGroup>

//         <Button variant="outlined" onClick={() => { alert('clicked') }}>Click me</Button>
//         <p>
//           Edit <code>src/App.js</code> and save to reload this bich.
//         </p>
//         <a
//           className="App-link"
//           href="https://reactjs.org"
//           target="_blank"
//           rel="noopener noreferrer"
//           >
//           Learn React
//         </a>
//         </header>
      
//     </div>
//     )
//   }



export default App;