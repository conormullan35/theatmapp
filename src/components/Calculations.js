import React, { useEffect } from "react";

const Calculations = (props) => {
  
  //Declare variables
  const five = 5;
  const ten = 10;
  const twenty = 20;

  //Declare variables
  const { state } = props;
  const { disp } = props;

  const value = disp["total"];


  //Parameters are amount - the amount requested and limits - the amount left in the atm
  const GetMoney = (amount, limits) => {
  //Greedy algorithm to select the distribution of notes based on the parameters supply and demand
    let recur = (amount, nominals) => {
      if (amount == 0) return {};
      if (!nominals.length) return;
      let nominal = nominals[0];
      let count = Math.min(limits[nominal], Math.floor(amount / nominal));
      for (let i = count; i >= 0; i--) {
        let result = recur(amount - i * nominal, nominals.slice(1));
        if (result)
          return i
            ? {
                [nominal]: i,
                ...result,
              }
            : result;
      }
    };
    //Recur function until total amount is reached
    return recur(
      amount,
      Object.keys(state)
        .map(Number)
        .sort((a, b) => b - a)
    );
  };

  //Loop through the atm note inventory and update the number of notes accordingly
  function adjustStock(output) {
    for (var key in state) {
      if (state.hasOwnProperty(key)) {
        for (var keyb in output) {
          if (output.hasOwnProperty(keyb)) {
            if (key == keyb) {
              state[key] = state[key] - output[keyb];
              if (state[key] <= 0) {
                delete state[key];
              }
            }
          }
        }
      }
    }
  }

  //Call the GetMoney function and store results as an array
  const returnval = GetMoney(value, state);

  //Methods to update the dispensed state based on the note value
  const setTw = () => {
    props.setDisp((prevState) => ({
      ...prevState,
      20: returnval[20],
    }));
  };
  const setTn = () => {
    props.setDisp((prevState) => ({
      ...prevState,
      10: returnval[10],
    }));
  };
  const setFv = () => {
    props.setDisp((prevState) => ({
      ...prevState,
      5: returnval[5],
    }));
  };

  //Once the state is changed, call the update functions to update the amount of notes dispensed
  useEffect(() => {
    if (returnval) {
      if (returnval[20] && returnval[20] != disp[twenty]) {
        setTw();
      }
      if (returnval[10] && returnval[10] != disp[ten]) {
        setTn();
      }
      if (returnval[5] && returnval[5] != disp[five]) {
        setFv();
      }
    }
  }, [state]);

  adjustStock(returnval);
  return null;
};

export default Calculations;
