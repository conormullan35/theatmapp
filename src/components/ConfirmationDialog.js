import React, { useEffect } from 'react';
import ReactDOM from 'react';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';

const ConfirmationDialog = (props) =>{

  //Import props
  const {balance} = props;
  const {setBalance} = props;
  const {value} = props;
  const {setIsLoading} = props;
  const {isDisabled} = props;
  const {setVisible} = props;
  const {showDialog} = props;
  const {setShowDialog} = props;
  const {setOpen} = props;
  const {open} = props;

  //Toggle values when state changes
  useEffect(()=>{
    if(open){
      setOpen(false);
    }
    if(!open && showDialog==true){
      setOpen(true);
    }

  }, [showDialog] );

  //Toggle values when close is clicked
  const handleClose = () => {
    setShowDialog(false);
    setOpen(false)
  };

  const handleOpen = () => {
    //Only allow withdrawl if within constraints of balance/overdraft
    if(balance-value>=-100){
        if(value%5===0){
          setIsLoading((prevLoading) => !prevLoading)
          setVisible((prevLoading) => !prevLoading)
          setBalance(prevBal => prevBal - value);
          setOpen(false)
          isDisabled(true)
        }
        else alert ('This machine only dispenses £5, £10 and £20 notes.');
      }
      //If exceeding the limits, throw an error
      if(balance-value<-100) alert('You have surpassed your overdraft amount');
      setShowDialog(false);
  }
  
  return (
    <React.Fragment>
      {props.children}
      {open ? (
        <div>
        <Dialog
          open={open}
          onClose={()=>handleClose()}
          aria-labelledby="alert-dialog-title"
          aria-describedby="alert-dialog-description"
        >
          <DialogTitle id="alert-dialog-title">{"Confirm Cash Withdrawl"}</DialogTitle>
          <DialogContent>
            <DialogContentText id="alert-dialog-description">
              Are you sure you want to withdraw £{props.value}?
            </DialogContentText>
          </DialogContent>
          <DialogActions>
            <Button onClick={()=>handleClose()} color="primary">
              Cancel
            </Button>
            <Button id='openBtn' onClick={()=>handleOpen()} color="primary" autoFocus>
              Confirm
            </Button>
          </DialogActions>
        </Dialog>
      </div>) : null}
    </React.Fragment>
  );
}

export default ConfirmationDialog;