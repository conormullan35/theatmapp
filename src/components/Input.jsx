import React from "react";
import "../css/Input.css";

const Input = (props) =>{

  //Import props
  const {len} = props;

  //Initially set all background to a transparent fill
  let elems = new Array(4).fill('transparent')

  //Depending on how many elements in string, fill each circle
  for(var i=0;i<len;i++){
    elems[i]='white';
  }

  return (
    //JSX for pin circles
    <div className="input">
      <div className="circle" id={'one'} style={{backgroundColor: elems[0]}}></div>
      <div className="circle" id={'two'} style={{backgroundColor: elems[1]}}></div>
      <div className="circle" id={'three'} style={{backgroundColor: elems[2]}}></div>
      <div className="circle" id={'four'} style={{backgroundColor: elems[3]}}></div>
    </div>
    );
  }

export default Input;