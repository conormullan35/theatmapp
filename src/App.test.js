import { render, screen } from "@testing-library/react";
import App from "./App";
import { shallow, mount } from "enzyme";
import React from "react";
import Calculations from "./components/Calculations";
import ConfirmationDialog from "./components/ConfirmationDialog";
import Dashboard from "./components/Dashboard";
import {confirmWithdraw} from "./components/Dashboard";
import Input from "./components/Input";
import NumPad from "./components/NumPad";
import PinCode from "./components/PinCode";
import SplashScreen from "./components/SplashScreen";
import Dialog from '@material-ui/core/Dialog';
import toJson from 'enzyme-to-json';

  const props = {
    initialProps: {
      showDialog: false,
    },
    dashboardProps: {
      showDialog: false,
    },
    calcProps: {
      showDialog: false,
      disp: { 20: 0, 10: 0, 5: 0, total: 25 },
      state: { 20: 7, 10: 15, 5: 4 },
    },
    dialogProps: {
      showDialog: true,
    },
    inputProps: {
      len: 0,
    },
    numProps: {
      leftname: "left",
      rightname: "right",
      disp: { 20: 0, 10: 0, 5: 0, total: 25 },
    },
  };

describe("Test rendering components and they accept passed props", () => {

  const container = shallow(<PinCode {...props.initialProps} />);
  it("renders App component without crashing", () => {
    shallow(<App />);
  });
  it("renders SplashScreen component header without crashing", () => {
    const wrapper = shallow(<SplashScreen />);
    const header = (
      <div className={"behind"}>Please insert your card to continue.</div>
    );
    expect(wrapper.contains(header)).toEqual(true);
  });
  it("renders PinCode component header without crashing", () => {
    expect(container.exists()).toBe(true);
  });

  const shDashboard = shallow(<Dashboard {...props.dashboardProps} />);
  it("renders Dashboard component without crashing", () => {
    expect(shDashboard.exists()).toBe(true);
  });

  const shCalculations = shallow(<Calculations {...props.calcProps} />);
  it("renders Calculations component without crashing", () => {
    expect(shCalculations.exists()).toBe(true);
  });

  const shDialog = shallow(<ConfirmationDialog {...props.dialogProps} />);
  it("renders Dialog component without crashing", () => {
    expect(shDialog.exists()).toBe(true);
  });

  const shInput = shallow(<Input {...props.inputProps} />);
  it("renders Input component without crashing", () => {
    expect(shInput.exists()).toBe(true);
  });

  const shNumPad = shallow(<NumPad {...props.numProps} />);
  it("renders NumPad component without crashing", () => {
    expect(shNumPad.exists()).toBe(true);
  });
});
const wrapper = mount(<Dashboard {...props.dashboardProps} />);
describe("logic", () => {
  
  it("button click - confirm amount", () => {
    wrapper.find(".btnsvn").simulate("click");
    wrapper.find(".confirmBtn").simulate("click");
    const inputValue = wrapper.find(".inputAmount").props().value;
    const expectedValue = "7.00";
    expect(inputValue).toEqual(expectedValue);
    wrapper.unmount();
  });
});

describe("click input buttons", () => {
  it("button click - confirm amount", () => {
    wrapper.mount();
    wrapper.find(".btnfv").simulate("click");
    wrapper.find(".btnfv").simulate("click");
    wrapper.find(".confirmBtn").simulate("click");
    const inputValue = wrapper.find(".inputAmount").props().value;
    const expectedValue = "55.00";
    expect(inputValue).toEqual(expectedValue);
    wrapper.unmount();
  });
});

describe("show dialog box", () => {
  it("when valid amount is entered, show dialog box", () => {
    const shConfirm = mount(<Dashboard {...props.dashboardProps}></Dashboard>);
    shConfirm.find(".btnfv").simulate("click");
    shConfirm.find(".btnfv").simulate("click");
    jest.mock('./components/Dashboard', () => ({ confirmWithdraw: jest.fn() }))
    shConfirm.find(".confirmBtn").simulate("click");
    expect(shConfirm.find(Dialog).length).toBe(1);
  });
});

describe("confirming amount and getting money", () => {
  it("should return notes images and hide the numberpad", () => {
    wrapper.mount();
    wrapper.find(".btnfv").simulate("click");
    wrapper.find(".btnfv").simulate("click");

    jest.mock('./components/Dashboard', () => ({ confirmWithdraw: jest.fn() }))
    const diWr = mount(<ConfirmationDialog {...props.ConfirmationDialog} />);
    wrapper.find(".confirmBtn").simulate("click");
    jest.mock('./components/ConfirmationDialog', () => ({ handleOpen: jest.fn() }))
    wrapper.find("#openBtn").first().simulate("click");
    expect(wrapper.find('img').length).toBe(3);
    wrapper.unmount();
  });
});

describe("confirming amount and getting money overdraft", () => {
  it("should return notes images and hide the numberpad", () => {
    wrapper.mount();
    wrapper.find(".btntw").simulate("click");
    wrapper.find(".btnnne").simulate("click");
    wrapper.find(".btnzr").simulate("click");
    jest.mock('./components/Dashboard', () => ({ confirmWithdraw: jest.fn() }))
    const diWr = mount(<ConfirmationDialog {...props.ConfirmationDialog} />);
    wrapper.find(".confirmBtn").simulate("click");
    jest.mock('./components/ConfirmationDialog', () => ({ handleOpen: jest.fn() }))
    wrapper.find("#openBtn").first().simulate("click");
    expect(wrapper.find('.ovrDrft').length).toBe(1);
    wrapper.unmount();
  });
});

describe("Enter a value not available given the notes in the atm (29)", () => {
  it("is not possible to dispense, throw error", () => {
    const wrEr = mount(<Dashboard {...props.initialProps}></Dashboard>);
    const jsdomAlert = window.alert;  // remember the jsdom alert
    window.alert = () => {}; 
    wrEr.find(".btntw").simulate("click");
    wrEr.find(".btnnne").simulate("click");
    jest.mock('./components/Dashboard', () => ({ confirmWithdraw: jest.fn() }))
    const diWr = mount(<ConfirmationDialog {...props.ConfirmationDialog} />);
    wrEr.find(".confirmBtn").simulate("click");
    jest.mock('./components/ConfirmationDialog', () => ({ handleOpen: jest.fn() }))
    wrEr.find("#openBtn").first().simulate("click");
    expect(wrEr.find('.slideInD').length).toBe(0);
    wrEr.unmount();
    window.alert = jsdomAlert; 
  });
});

describe("test pin logic", () => {
  it("should not expect a value other than 1234", () => {
    const pc = mount(<PinCode {...props.initialProps} />);
    const jsdomAlert = window.alert;  // remember the jsdom alert
    window.alert = () => {}; 
    pc.mount();
    pc.find(".btntw").simulate("click");
    pc.find(".btnnne").simulate("click");
    pc.find(".btnzr").simulate("click");
    jest.mock('./components/PinCode', () => ({ verifyPin: jest.fn() }))
    pc.find(".confirmBtn").first().simulate("click");
    expect(pc.props().showDialog).toBe(false);
    pc.unmount();
    window.alert = jsdomAlert; 
  });
});

describe("snapshots", () => {
  it("App snapshot", () => {
    const tree = shallow(<App/>);
    expect(toJson(tree)).toMatchSnapshot();
  });
  it("Calculations snapshots", () => {
    const clSh = shallow(<Calculations {...props.calcProps} />);
    expect(toJson(clSh)).toMatchSnapshot();
  });
  it("Dialog snapshot", () => {
    const cdSh = shallow(<ConfirmationDialog />);
    expect(toJson(cdSh)).toMatchSnapshot();
  });
  it("Dashboard snapshots", () => {
    const dbSh = shallow(<Dashboard/>);
    expect(toJson(dbSh)).toMatchSnapshot();
  });
  it("Input snapshot", () => {
    const inSh = shallow(<Input />);
    expect(toJson(inSh)).toMatchSnapshot();
  });
  it("NumPad snapshots", () => {
    const npSh = shallow(<NumPad/>);
    expect(toJson(npSh)).toMatchSnapshot();
  });
  it("PinCode snapshot", () => {
    const pcdSh = shallow(<PinCode />);
    expect(toJson(pcdSh)).toMatchSnapshot();
  });
  it("SplashScreen snapshot", () => {
    const ssSh = shallow(<SplashScreen />);
    expect(toJson(ssSh)).toMatchSnapshot();
  });
});