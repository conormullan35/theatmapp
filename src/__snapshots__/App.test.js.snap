// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots App snapshot 1`] = `
<main>
  <WithStyles(ForwardRef(Container))
    maxWidth="sm"
  >
    <Switch>
      <Route
        exact={true}
        path="/"
      >
        <Redirect
          component={[Function]}
          to="/splashscreen"
        />
      </Route>
      <Route
        component={[Function]}
        path="/splashscreen"
      />
      <Route
        component={[Function]}
        path="/dashboard"
      />
      <Route
        component={[Function]}
        path="/pincode"
      />
    </Switch>
  </WithStyles(ForwardRef(Container))>
</main>
`;

exports[`snapshots Calculations snapshots 1`] = `""`;

exports[`snapshots Dashboard snapshots 1`] = `
<Fragment>
  <h2
    className="name"
  >
    Welcome Michael
  </h2>
  <div
    id="dashboardWrap"
  >
    <div
      className="topWrap"
    >
      <div
        className="three"
      >
        <div
          className="ca first"
        >
          <p>
            Balance
          </p>
          <p
            className="snd white"
          >
            £
            220
            .00
          </p>
        </div>
        <div
          className="ca last"
        >
          <p>
            Overdraft Limit
          </p>
          <p
            className="snd "
          >
            £100.00
          </p>
        </div>
      </div>
    </div>
    <div
      className="twoCards"
    />
    <form
      id="formid"
      onSubmit={[Function]}
    >
      <input
        className="inputAmount"
        disabled={false}
        name="total"
        onChange={[Function]}
        readOnly={true}
        style={
          Object {
            "fontFamily": "BebasNeue-Regular",
          }
        }
        type="numeric"
        value=".00"
      />
    </form>
    <div
      style={
        Object {
          "backgroundColor": "transparent",
        }
      }
    >
      <NumPad
        disp={
          Object {
            "10": 0,
            "20": 0,
            "5": 0,
            "total": "",
          }
        }
        leftAction={[Function]}
        leftName={<Memo />}
        open={false}
        rightAction={[Function]}
        rightName={
          <ConfirmationDialog
            balance={220}
            isDisabled={[Function]}
            isLoading={false}
            limits={[Function]}
            open={false}
            setBalance={[Function]}
            setIsLoading={[Function]}
            setOpen={[Function]}
            setShowDialog={[Function]}
            setVisible={[Function]}
            showDialog={false}
            value=""
          >
            <Memo />
          </ConfirmationDialog>
        }
        setDisp={[Function]}
        setOpen={[Function]}
        setShowDialog={[Function]}
        showDialog={false}
      />
      <div
        className="row"
      >
        <button
          className="withdrBtn cancel"
          onClick={[Function]}
        >
          CANCEL TRANSACTION
        </button>
      </div>
    </div>
    <div
      className="bN"
    />
  </div>
</Fragment>
`;

exports[`snapshots Dialog snapshot 1`] = `<Fragment />`;

exports[`snapshots Input snapshot 1`] = `
<div
  className="input"
>
  <div
    className="circle"
    id="one"
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
  />
  <div
    className="circle"
    id="two"
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
  />
  <div
    className="circle"
    id="three"
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
  />
  <div
    className="circle"
    id="four"
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
  />
</div>
`;

exports[`snapshots NumPad snapshots 1`] = `
<div
  className="app"
>
  <div
    className="calc-wrapper"
  >
    <div
      className="row"
    >
      <button
        className="button-wrapper btnsvn"
        onClick={[Function]}
      >
        7
      </button>
      <button
        className="button-wrapper btnegt"
        onClick={[Function]}
      >
        8
      </button>
      <button
        className="button-wrapper btnnne"
        onClick={[Function]}
      >
        9
      </button>
    </div>
    <div
      className="row"
    >
      <button
        className="button-wrapper btnfr"
        onClick={[Function]}
      >
        4
      </button>
      <button
        className="button-wrapper btnfv"
        onClick={[Function]}
      >
        5
      </button>
      <button
        className="button-wrapper btnsx"
        onClick={[Function]}
      >
        6
      </button>
    </div>
    <div
      className="row"
    >
      <button
        className="button-wrapper btnon"
        onClick={[Function]}
      >
        1
      </button>
      <button
        className="button-wrapper btntw"
        onClick={[Function]}
      >
        2
      </button>
      <button
        className="button-wrapper btnthr"
        onClick={[Function]}
      >
        3
      </button>
    </div>
    <div
      className="row"
    >
      <button
        className="button-wrapper clearBtn"
        onClick={[Function]}
      />
      <button
        className="button-wrapper btnzr"
        onClick={[Function]}
      >
        0
      </button>
      <button
        className="button-wrapper confirmBtn"
        onClick={[Function]}
      />
    </div>
  </div>
</div>
`;

exports[`snapshots PinCode snapshot 1`] = `
<Fragment>
  <div
    className="behind"
  >
    Please enter your PIN to continue.
  </div>
  <div
    className="app"
  >
    <div
      className="calc-wrapper "
    >
      <Input
        fill="transparent"
        input=""
        len={0}
        type="password"
      />
      <div
        className="row"
      />
      <NumPad
        disp={
          Object {
            "total": "",
          }
        }
        leftAction={[Function]}
        leftName={<Memo />}
        rightAction={[Function]}
        rightName={<Memo />}
        setDisp={[Function]}
      />
    </div>
  </div>
</Fragment>
`;

exports[`snapshots SplashScreen snapshot 1`] = `
<div
  className="root"
>
  <div
    className="behind"
  >
    Please insert your card to continue.
  </div>
  <div
    className="cardSlot"
  />
  <div
    className="cardDiv"
  >
    <WithStyles(ForwardRef(Collapse))
      direction="down"
      in={true}
      mountOnEnter={true}
      unmountOnExit={true}
    >
      <div>
        <button
          className="cardBtn"
          onClick={[Function]}
        >
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  576,
                  512,
                  Array [],
                  "f09d",
                  "M0 432c0 26.5 21.5 48 48 48h480c26.5 0 48-21.5 48-48V256H0v176zm192-68c0-6.6 5.4-12 12-12h136c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H204c-6.6 0-12-5.4-12-12v-40zm-128 0c0-6.6 5.4-12 12-12h72c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12v-40zM576 80v48H0V80c0-26.5 21.5-48 48-48h480c26.5 0 48 21.5 48 48z",
                ],
                "iconName": "credit-card",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          />
        </button>
      </div>
    </WithStyles(ForwardRef(Collapse))>
  </div>
</div>
`;
